tools/hardware/utils/memory.py

```diff
@@ -83,12 +83,31 @@ def reserve_regions(regions: Set[Region], reserved: Set[Region]) -> Set[Region]:
     return ret
 
 
+def align_memory(regions: Set[Region], config: Config) -> List[Region]:
+    ''' Given a set of regions, sort them and align the first so that the
+    ELF loader will be able to load the kernel into it. Will return the
+    aligned memory region list, a set of any regions of memory that were
+    aligned out and the physBase value that the kernel will use. '''
+
+    ret = sorted(regions)
+    extra_reserved = set()
+
+    if config.get_kernel_phys_align() != 0:
+        new = ret[0].align_base(config.get_kernel_phys_align())
+        resv = Region(ret[0].base, new.base - ret[0].base)
+        extra_reserved.add(resv)
+        ret[0] = new
+
+    physBase = ret[0].base
+    return ret, extra_reserved, physBase
+
+
 def get_physical_memory(tree: FdtParser, config: Config) -> List[Region]:
     ''' returns a list of regions representing physical memory as used by the kernel '''
     regions = merge_memory_regions(get_memory_regions(tree))
     reserved = parse_reserved_regions(tree.get_path('/reserved-memory'))
     regions = reserve_regions(regions, reserved)
-    regions, extra_reserved, physBase = config.align_memory(regions)
+    regions, extra_reserved, physBase = align_memory(regions, config)
```

Comment:
Wouldn't is make sense to inline `align_memory() here, because it is doing something really special that is only used here?

Reply:
I was just reverting it back to how it was before it was refactored into the Config class.  The other functions called by get_physical_memory are helpers defined in this file, so it doesn't seem like an issue for align_memory to continue to be one too.